### PR TITLE
Adding API support to fetch templates assigned to resource

### PR DESCRIPTION
**Pull request by VigneshOK10**
https://hpe-dev.docs.try.opsramp.com/api/monitoring/monitoring-templates-assigned-templates-get/

### CODE REVIEW

File: opsramp/resources.py [hunk 1]
```diff
@@ -89,3 +89,9 @@ def availability(self, uuid, start_epoch, end_epoch):
             uuid, start_epoch, end_epoch
         )
         return self.api.get(url_suffix)
+
+    def get_templates(self, uuid, pattern=''):
+        url_suffix = '{0}/templates/search?{1}'.format(
+            uuid, pattern
+        )
+        return self.api.get(url_suffix)
```
Comment (jofegan) [inline opsramp/resources.py:97]: Calling api.get will work as long as there is only one page of results. However I suggest changing to the following, which takes advantage of the superclass's support for paginated search results.
```
def get_templates(self, uuid, pattern=None):
     url_suffix = '{0}/templates'.format(uuid)
     self.api.pushd(url_suffix)
     resp = super(Resources, self).search(pattern)
     resp = self.search(pattern)
     self.api.popd()
     return resp
```

**Review by jofegan — CHANGES_REQUESTED** [review]
Looks good, just an improvement so that it will handle multi-page responses if a resource has a large set of templates attached to it.

File: opsramp/resources.py (hunk 1 above)
Comment (VigneshOK10) [reply opsramp/resources.py]: thanks Josie. Fixed it

**Review by jofegan — APPROVED** [review]
LGTM

**Pull request merged**